docs/content/latest/api/ysql/the-sql-language/statements/ddl_truncate.md

```diff
@@ -43,52 +45,125 @@ Use the `TRUNCATE` statement to clear all rows in a table.
+drop table if exists parents  cascade;
+
+create table parents(k int primary key, v text not null);
+
+create table children(
+  parent_k  int  not null,
+  k         int  not null,
+  v         text not null,
+
+  constraint children_pk primary key(parent_k, k),
+
+  constraint children_fk foreign key(parent_k)
+    references parents(k)
+    match full
+    on delete cascade
+    on update restrict);
+
+insert into parents(k, v) values (1, 'dog'), (2, 'cat'), (3, 'frog');
+
+insert into children(parent_k, k, v) values
+  (1, 1, 'dog-child-a'),
+  (1, 2, 'dog-child-b'),
+  (1, 3, 'dog-child-c'),
+  (2, 1, 'cat-child-a'),
+  (2, 2, 'cat-child-b'),
+  (2, 3, 'cat-child-c'),
+  (3, 1, 'frog-child-a'),
+  (3, 2, 'frog-child-b'),
+  (3, 3, 'frog-child-c');
+
+select p.v as "parents.v", c.v as "chilgren.v"
+from parents p inner join children c on c.parent_k = p.k
+order by p.k, c.k;
 ```
 
+This is the result:
+
+```output
+ parents.v |  chilgren.v  
+-----------+--------------
+ dog       | dog-child-a
+ dog       | dog-child-b
+ dog       | dog-child-c
+ cat       | cat-child-a
+ cat       | cat-child-b
+ cat       | cat-child-c
+ frog      | frog-child-a
+ frog      | frog-child-b
+ frog      | frog-child-c
 ```
- k1 | k2 | v1 | v2
-----+----+----+----
-  1 |  2 |  3 | a
-  2 |  3 |  4 | b
-  3 |  4 |  5 | c
-(3 rows)
+
+The `\d children` metacommand shows that it has a foreign key constraint to the  `parents` table.  This makes it a (transitive) dependent object of that table:
+
+```output
+Indexes:
+    "children_pk" PRIMARY KEY, lsm (parent_k HASH, k ASC)
+Foreign-key constraints:
+    "children_fk" FOREIGN KEY (parent_k) REFERENCES parents(k) MATCH FULL ON UPDATE RESTRICT ON DELETE CASCADE
 ```
+Notice that the effect of the `on delete cascade` clause is limited to what the `delete` statement does. It has no effect on the behavior of `truncate`. (There is no `on truncate cascade` clause.) Try `delete from parents`. It quietly succeeds and removes all the rows from both the `parents` table and the `children` table.
+
+With all the rows that the setup code above inserts, try this:
 
 ```plpgsql
-yugabyte=# TRUNCATE sample;
+do $body$
+declare
+  message  text not null := '';
+  detail   text not null := '';
+begin
+  -- Causes error 'cos "cascade" is required.
+  truncate table parents;
+  assert false, 'Should not get here';
+exception
+  -- Error 0A000
+  when feature_not_supported then
+    get stacked diagnostics
+      message  = message_text,
+      detail   = pg_exception_detail;
+    assert message = 'cannot truncate a table referenced in a foreign key constraint',  'Bad message';
+    assert detail  = 'Table "children" references "parents".',                          'Bad detail';
+end;
+$body$;
 ```
 
+It finishes without error, showing that the bare `truncate table parents`, without `cascade`, fails and causes the message and hint that the code presents. Now repeat the attempt with `cascade` and observe the result:
```

Comment:
Same suggestion as the previous version of this paragraph...

Reply:
I don't know what you mean. So I'm leaving it as is.